streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -607,8 +613,12 @@ private void prepareTopic(Map<String, InternalTopicMetadata> topicPartitions) {
                 InternalTopicConfig topic = entry.getValue().config;
                 Integer numPartitions = entry.getValue().numPartitions;
 
-                if (numPartitions < 0)
+                if (numPartitions == NOT_AVAILABLE) {
+                    continue;
+                }
+                if (numPartitions < 0) {
```

Comment:
This could be `UNKOWN` or `NOT_AVAILABLE`, thus `< 0` does check for both at once. I could change it to `(numPartitions == UNKNOWN || numPartitions == NOT_AVAILABLE)` of course... WDYT?